.github/workflows/unit-test.yml

```
@@ -0,0 +1,58 @@
+name: Unit Tests
+
+on: [push, pull_request]
+
+jobs:
+  build:
```

The job will be identified by `build`, so in this case it's better to rename it to `tests`

```suggestion
  build:
```